src/coreclr/vm/tieredcompilation.cpp

```diff
@@ -1016,9 +1088,24 @@ CORJIT_FLAGS TieredCompilationManager::GetJitFlags(PrepareCodeConfig *config)
         NativeCodeVersion::OptimizationTier newOptimizationTier;
         if (!methodDesc->RequestedAggressiveOptimization())
         {
+            NativeCodeVersion::OptimizationTier currentTier = nativeCodeVersion.GetOptimizationTier();
+            if (currentTier == NativeCodeVersion::OptimizationTier::OptimizationTierInstrumented)
+            {
+                flags.Set(CORJIT_FLAGS::CORJIT_FLAG_BBINSTR);
+                flags.Set(CORJIT_FLAGS::CORJIT_FLAG_TIER0);
+                return flags;
+            }
+
+            if (currentTier == NativeCodeVersion::OptimizationTier::OptimizationTierInstrumentedOptimized)
+            {
+                flags.Set(CORJIT_FLAGS::CORJIT_FLAG_BBINSTR);
+                flags.Set(CORJIT_FLAGS::CORJIT_FLAG_TIER1);
```

Comment:
After I addressed all of your feedback it now sends TIER1 flag to instrumented tier after R2R (only if a corresponding strategy was selected)

Reply:
I meant whether it would make sense to not send the tier 1 flag for even the optimized instrumented tier. It would still be optimized without that flag. Wondering whether the tier 1 flag may in the future enable more optimizations that use extra CPU time, which may not be worthwhile for an instrumented tier.

Reply:
Ok, going to replace FLAG_TIER1 with FLAG_OPT here, for now both will have the same effects but at least we won't be printing "Tier1" for JitDisasm for instrumented tiers in the comments